mos-platform/osi-c1p/putchar.cc

```diff
@@ -0,0 +1,30 @@
+#include "osi_screen.h"
+
+template<unsigned int scr_base_int, unsigned int video_ram_size,
+    unsigned int screen_width, unsigned int screen_height,
+    unsigned int screen_firstchar, unsigned int scroll_dist>
+    unsigned char
+    osi_screen<scr_base_int, video_ram_size, screen_width, screen_height,
+         screen_firstchar, scroll_dist>::cursor_x;
+
+template<unsigned int scr_base_int, unsigned int video_ram_size,
+    unsigned int screen_width, unsigned int screen_height,
+    unsigned int screen_firstchar, unsigned int scroll_dist>
+    unsigned char
+    osi_screen<scr_base_int, video_ram_size, screen_width, screen_height,
+         screen_firstchar, scroll_dist>::cursor_y;
+
+using c1p_screen = osi_screen<>;
+
+/**
+ * @brief __putchar implementation for Challenger 1P
+ * 
+ * For other screen layouts this module must be overridden.
+ * 
+ * @param c 
+ */
+extern "C"
+void __putchar(char c)
```

Comment:
See the C standard: https://port70.net/~nsz/c/c99/n1256.html#5.2.2

In particular, for C, the expectation for '\n' is that it should move to the *initial* position on the next line, while '\r' moves to the initial position on the *current* line. This is in conflict with the traditional use of CR and LF on printers and such, but much C code expects it.

Typcially, CRLF systems add a translation layer under the C standard interfaces (here, backed by `__putchar`) to emit the right sequence of underlying control codes. For your target, you could either translate here, or break from the usual conio convention in `cputc`. I think it'd probably be a good idea to eventually support a CC65-compatible conio, so translating seems preferable to me, but your call.